data_structures/linked_list/quicksort_linkedlist.py

```diff
@@ -0,0 +1,91 @@
+"""
+Given a linked list with head pointer,
+sort the linked list using quicksort technique without using any extra space
+Time complexity: O(NlogN), Space complexity: O(1)
+"""
+from __future__ import annotations
+
+
+class Node:
+    def __init__(self, data: int):
+        self.data = data
+        self.next = None
+
+
+class LinkedList:
+    def __init__(self):
+        self.head = None
+
+    # method to insert nodes at the start of linkedlist
+    def insert(self, new_data: int):
+        """
+        Pushes the element at the start of the linkedlist and returns None
+        """
+        new_node = Node(new_data)
+        new_node.next = self.head
+        self.head = new_node
+        return
+
+    # method to print the linkedlist
+    def printll(self):
+        """
+        Prints the linkedlist in the particular format and returns None
+        example: 1 ->3 ->4 ->2 ->9 ->6 ->8 -> 7
+        """
+        temp = self.head
+        if temp == None:
+            return "Linked List is empty"
```

Comment:
```suggestion
            return
```
Returning `str` is a bad idea - Either raise an error, return, or print that it is empty